packages/typir/src/features/operator.ts

```diff
@@ -22,35 +22,53 @@ export type InferOperatorWithMultipleOperands<T = unknown> = {
     operands: (domainElement: T, operatorName: string) => unknown[];
 };
 
+export interface UnaryOperatorDetails<T> {
+    name: string;
+    signature: UnaryOperatorSignature | UnaryOperatorSignature[];
+    inferenceRule?: InferOperatorWithSingleOperand<T>;
+}
+export interface UnaryOperatorSignature {
+    operand: Type;
+    return: Type;
+}
+
+export interface BinaryOperatorDetails<T> {
+    name: string;
+    signature: BinaryOperatorSignature | BinaryOperatorSignature[];
+    inferenceRule?: InferOperatorWithMultipleOperands<T>;
+}
+export interface BinaryOperatorSignature {
+    left: Type;
+    right: Type;
+    return: Type;
+}
+
+export interface TernaryOperatorDetails<T> {
+    name: string;
+    signature: TernaryOperatorSignature | TernaryOperatorSignature[];
+    inferenceRule?: InferOperatorWithMultipleOperands<T>;
+}
+export interface TernaryOperatorSignature {
+    first: Type;
+    second: Type;
+    third: Type;
+    return: Type;
+}
+
+export interface GenericOperatorDetails<T> {
+    name: string;
+    outputType: Type;
+    inputParameter: NameTypePair[];
+    inferenceRule?: InferOperatorWithSingleOperand<T> | InferOperatorWithMultipleOperands<T>;
+}
+
 export interface OperatorManager {
```

Comment:
OperatorFactory? Manager sounds like too many responsibilities

Reply:
You are right. Initially, I expected, that we would have more responsibilities for operators beyond creating operators. I added a TODO in the code to revise the name later again.